api/namex/resources/name_requests/name_request.py

```diff
@@ -250,72 +259,152 @@ def handle_patch_edit(self, nr_model):
+            payment_info = payment_request.get('paymentInfo')
+            filing_info = payment_request.get('filingInfo')
+            business_info = payment_request.get('businessInfo')
 
-        # This handles the updates for NRO and Solr, if necessary
-        # TODO: Do we update network services?
-        self.update_records_in_network_services(nr_model)
-        return nr_model
+            # Create our payment request
+            req = PaymentRequest(
+                payment_info=payment_info,
+                filing_info=filing_info,
+                business_info=business_info
+            )
 
-    def handle_patch_refund(self, nr_model):
-        # This handles updates if the NR state is 'patchable'
-        nr_model = self.update_nr_fields(nr_model, nr_model.stateCd)
+            payment = create_payment(req)
+            if not payment:
+                raise PaymentServiceException(MSG_ERROR_CREATING_RESOURCE)
+
+            # Update the name request with the payment id
+            # nr_draft.paymentToken = str(payment.id)
+            nr_model.payment_token = str(payment.id)
+            # nr_model.payment_completion_date = ''
 
+            nr_model.priorityCd = 'Y'
+            # nr_model.priorityDate = ''
+
+            # Save the name request
+            nr_model.save_to_db()
+
+        except Exception as err:
+            raise NameRequestException(err, message='Error upgrading Name Request!')
+
+        # We have not accounted for multiple payments.
+        # We will need to add a request_payment model (request_id and payment_id)
         # This handles the updates for NRO and Solr, if necessary
-        # self.update_records_in_network_services(nr_model)
+        self.update_records_in_network_services(nr_model)
         return nr_model
 
     def handle_patch_reapply(self, nr_model):
+        """
+        Extend the Name Request's expiration date by 56 days. If the request action is set to REH or REST,
+        extend the expiration by an additional year (plus the default 56 days).
+        :param nr_model:
+        :return:
+        """
         nr_svc = self.nr_service
 
-        # This handles updates if the NR state is 'patchable'
-        # TODO: Ensure request action is REH or REST <- Where does this come in?
-        # TODO: Ensure submit count is not greater than 3
-
         if nr_model.submitCount < 3:
-            # Update submit count
-            nr_model = nr_svc.update_request_submit_count(nr_model)
-            # Extend expiry date by (default) 56 days
-            nr_model = nr_svc.extend_expiry_date(nr_model)
-            # TODO: If request action is REH or REST extend by 1 year (+ 56 default) days
-            # TODO: Where does the request action come from in this scenario?
-            # nr_model = nr_svc.extend_expiry_date(nr_model, (datetime.utcnow() + relativedelta(years=1)))
+            if nr_svc.request_action in [RequestAction.REH.value, RequestAction.REST.value]:
```

Comment:
sorry the request_actions are REH(Restore from a Historical Name) and REN(Restore with a New Name). Please update.  I double-checked the front-end code.